lib/transrate/snap.rb

```diff
@@ -138,7 +138,8 @@ def build_index file, threads
           runner.run
           if !runner.status.success?
             err = runner.stderr
-            if err =~ /Ran out of overflow table namespace/
+            if err =~ /Ran out of overflow table namespace/ || err =~ /Trying to use too many overflow entries/
+              logger.warn "Snap index build failed with n = #{n} , increasing +1"
               n += 1
               Dir.delete(@index_name) if Dir.exist?(@index_name)
```

Comment:
The issue with the index that @jorvis and I had (issue #189 ) seems to stem from the directory with the index not being removed because there are files within it. So I think if we replace 

```
Dir.delete(@index_name) if Dir.exist?(@index_name)
```
with 
```
require 'fileutils'

FileUtils.rm_rf(@index_name) if Dir.exist?(@index_name)
```
or a safer option(?)
```
FileUtils.rmdir(@index_name) if Dir.exist?(@index_name)
```

it should do what is needed and actually remove the directory so it can rebuild the larger Snap index.